File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    
    function findRow(a,b,d,target){  
        while(a<=b){
            let m=parseInt((a+b)/2)
            // console.log("a: "+a+", b: "+b+", m: "+m)
            if(matrix[m][d]===target) return m
            
            target<matrix[m][d] ? b=m-1 : a=m+1
        }
        return a
    } 
    function findCol(a,b,d,target){  
        while(a<=b){
            let m=parseInt((a+b)/2)
            // console.log("a: "+a+", b: "+b+", m: "+m)
            if(matrix[d][m]===target) return m
            
            target<matrix[d][m] ? b=m-1 : a=m+1
        }
        return a
    } 
    
    let row=findRow(0, matrix.length-1,matrix[0].length-1,target)
    if(row>=matrix.length || row<0) 
        return false
    
    let col=findCol(0,matrix[0].length-1, row, target)
    if(col>=matrix[0].length || col<0) 
        return false
    
    return matrix[row][col]===target
    
};

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
import numpy
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        r=len(matrix)
        c=len(matrix[0])
        def helper(a,b, arr, target):
            while(a<=b):
                m=floor((a+b)/2)
                #  print("a: "+a+", b: "+b+", m: "+m)
                if(arr[m]==target):
                    return m
            
                if(target<arr[m]):
                    b=m-1
                else:
                    a=m+1
        
            return a
        
        rotated=numpy.rot90(matrix)
        # print(rotated[0])
        
        row=helper(0, r-1, rotated[0], target)
       
        if(row>=r or row<0):
            return False
        
        col=helper(0,c-1, matrix[row], target)
        if(col>=c or col<0):
            return False  
        
        if(matrix[row][col]==target):
            return True 
        else:
            return False